src/p11_child/p11_child_openssl.c

```diff
@@ -1991,7 +1991,35 @@ errno_t do_card(TALLOC_CTX *mem_ctx, struct p11_ctx *p11_ctx,
     /* login: do we need to check for Login Required? */
     if (mode == OP_AUTH) {
         DEBUG(SSSDBG_TRACE_ALL, "Login required.\n");
-        if (pin != NULL) {
+
+        /* Check for protected authentication path */
+        rv = modules[c]->C_GetTokenInfo(slot_id, &token_info);
```

Comment:
Hi,

the `token_info` collected earlier in the `for (;;)` loop should still be valid. Is there a reason you read it again?

bye,
Sumit

Reply:
Hi,
No reason, except that I was unsure if it is guaranteed that `token_info` is still valid at this point.
Regards
/Mathias

Reply:
Thanks